### ISSUE TITLE
False expression: Node must have a real position for this operation.

### ISSUE THREAD

**Issue by NullVoxPopuli** [typed-ember/glint#320]
When importing a gts file, I get this error:
```
❯ node_modules/.bin/glint
app/components/demo.gts:1:21 - error TS2306: File '<repo>/app/components/example.ts' is not a module.

1 import Example from './example';
                      ~~~~~~~~~~~
```

deps:
```
❯ pnpm list *glint*

devDependencies:
@glint/core 0.8.0
@glint/environment-ember-loose 0.8.0
glint-environment-ember-template-imports 0.0.4
```

repro: https://github.com/NullVoxPopuli/repro-glint-gts-error-parse-node
```
git clone git@github.com:NullVoxPopuli/repro-glint-gts-error-parse-node.git
cd repro-glint-gts-error-parse-node
git checkout ad965d3
node_modules/.bin/glint
```

The code that produces the error:
```
// demo.gts
import Example from './example';

const Demo = <template>
  <Example />
</template>

export default Demo;
```

```gts
// example.gts
<template>
  <div ...attributes>

  </div>
</template>

```

```hbs
{{!-- templates/application.hbs --}}

<Demo />
{{outlet}}
```

Likely related error:
```
app/components/demo.gts:1:21 - error TS2306: File '<repo>/app/components/example.ts' is not a module.

1 import Example from './example';
                      ~~~~~~~~~~~
```

I also noticed that if I do a normal gts export, glint doesn't complaint about the example import or file

```gts
// demo.gts
import Example from './example';

<template>
  <Example />
</template>
```
just gives this now (which is expected)
```
❯ node_modules/.bin/glint
app/templates/application.hbs:1:1 - error TS7053: Element implicitly has an 'any' type because expression of type '"Demo"' can't be used to index type 'Globals'.
  Property 'Demo' does not exist on type 'Globals'.

1 <Demo />
  ~~~~~~~~}

```
